internal/handlers/msg_service_router_test.go

```diff
@@ -1856,6 +1856,232 @@ func TestRewardsProgramStartPerformQualifyingActions_Vote_ValidDelegations(t *te
+
+	ctx := app.BaseApp.NewContext(false, tmproto.Header{})
+	ctx.WithBlockTime(time.Now())
+	app.AccountKeeper.SetParams(ctx, authtypes.DefaultParams())
+	require.NoError(t, testutil.FundAccount(app.BankKeeper, ctx, acct1.GetAddress(), sdk.NewCoins(sdk.NewCoin(msgfeestypes.NhashDenom, sdk.NewInt(290500010)))))
+	require.NoError(t, testutil.FundAccount(app.BankKeeper, ctx, acct3.GetAddress(), sdk.NewCoins(sdk.NewCoin(msgfeestypes.NhashDenom, sdk.NewInt(290500010)))))
+	coinsPos := sdk.NewCoins(sdk.NewInt64Coin(sdk.DefaultBondDenom, 100000000))
+	msg, err := govtypesv1beta1.NewMsgSubmitProposal(
+		ContentFromProposalType("title", "description", govtypesv1beta1.ProposalTypeText),
+		coinsPos,
+		addr3,
+	)
+
+	fees := sdk.NewCoins(sdk.NewInt64Coin("atom", 150))
+	ctx.WithBlockTime(time.Now())
+	acct3 = app.AccountKeeper.GetAccount(ctx, acct3.GetAddress()).(*authtypes.BaseAccount)
+	seq := acct3.Sequence
+	ctx.WithBlockTime(time.Now())
+	time.Sleep(200 * time.Millisecond)
+
+	app.BeginBlock(abci.RequestBeginBlock{Header: tmproto.Header{Height: 2, Time: time.Now().UTC()}})
+	txGov, err := SignTx(NewTestRewardsGasLimit(), sdk.NewCoins(sdk.NewInt64Coin("atom", 150), sdk.NewInt64Coin(msgfeestypes.NhashDenom, 1_190_500_000)), encCfg, priv3.PubKey(), priv3, *acct3, ctx.ChainID(), msg)
+	require.NoError(t, err)
+
+	_, res, errFromDeliverTx := app.SimDeliver(encCfg.TxConfig.TxEncoder(), txGov)
+	require.NoError(t, errFromDeliverTx)
+
+	app.EndBlock(abci.RequestEndBlock{Height: 2})
+	app.Commit()
+
+	seq = seq + 1
+	proposal := app.GovKeeper.GetProposals(ctx)
+
+	require.NotEmpty(t, proposal, "proposal has to exist")
+	// tx with a fee associated with msg type and account has funds
+	vote1 := govtypesv1beta1.NewMsgVote(addr3, proposal[0].Id, govtypesv1beta1.OptionYes)
+
+	assert.NotEmpty(t, res.GetEvents(), "should have emitted an event.")
+
+	// threshold will be met after 10 actions
+	for height := int64(3); height < int64(23); height++ {
+		app.BeginBlock(abci.RequestBeginBlock{Header: tmproto.Header{Height: height, Time: time.Now().UTC()}})
+		require.NoError(t, acct3.SetSequence(seq))
+		tx1, err1 := SignTx(NewTestRewardsGasLimit(), fees, encCfg, priv3.PubKey(), priv3, *acct3, ctx.ChainID(), vote1)
+		require.NoError(t, err1)
+		_, res, errFromDeliverTx := app.SimDeliver(encCfg.TxConfig.TxEncoder(), tx1)
+		require.NoError(t, errFromDeliverTx)
+		assert.NotEmpty(t, res.GetEvents(), "should have emitted an event.")
+		app.EndBlock(abci.RequestEndBlock{Height: height})
+		app.Commit()
+		seq = seq + 1
+	}
+	claimPeriodDistributions, err := app.RewardKeeper.GetAllClaimPeriodRewardDistributions(ctx)
+	require.NoError(t, err)
+	assert.Len(t, claimPeriodDistributions, 1, "claim period reward distributions should exist")
+	assert.Equal(t, int64(10), claimPeriodDistributions[0].TotalShares, "claim period has not ended so shares have to be 0")
```

Comment:
Same as above. Is this and the following messages correct? "claim period has ended so shares have to be 0".